### PR TITLE
[NCL-2211] Add new option to show difference of two products

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

File: cli-wrap/src/main/resources/cli/da-cli-admin.sh [hunk 1]
```diff
@@ -116,6 +116,10 @@ printUsage() {
     echo "        <groupId>:<artifactId>:<version> ::"
     echo "          <groupId>:<artifactId>:<version> <Exact Matched Red Hat Version> <In black/white list?> <Available Versions>"
     echo ""
+    echo "    $0 difference [--json] leftID rightID"
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli-admin.sh]: Test of new review process

File: cli-wrap/src/main/resources/cli/da-cli.sh [hunk 2]
```diff
@@ -91,6 +91,10 @@ printUsage() {
     echo "        <groupId>:<artifactId>:<version> ::"
     echo "          <groupId>:<artifactId>:<version> <Exact Matched Red Hat Version> <In black/white list?> <Available Versions>"
     echo ""
+    echo "    $0 difference [--json] leftID rightID"
+    echo "        Returns difference between two products given by their IDs"
+    echo "          --json                    Output unparsed response from Dependency Analyzer."
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli.sh]: Another review comment

**Comment by stanlyDoge** [comment]
retest this please

File: cli-wrap/src/main/resources/cli/pretty.py [hunk 3]
```diff
@@ -125,6 +125,69 @@ def printReportAlign(data):
...
+
+    if len(data['added']) == 0:
+        print "added : NONE"
+    else:
+        print "added :"
+        for module in data['added']:
+            print "    groupId : " + module['groupId'] 
+            print "    artifactId : " + module['artifactId']
+            print "    version : " + module['version']
+
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/pretty.py]: Here isn't blankline separator?

File: cli-wrap/src/main/resources/cli/pretty.py [hunk 4]
```diff
@@ -125,6 +125,69 @@ def printReportAlign(data):
...
+    if len(data['changed']) == 0:
+        print "changed : NONE"
+    else:
+        print "changed :"
+    for module in data['changed']:
+        print "    groupId : " + module['groupId'] 
+        print "    artifactId : " + module['artifactId']
+        print "    leftVersion : " + module['leftVersion']
+        print "    rightVersion : " + module['rightVersion']
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/pretty.py]: Here isn't blankline separator?

File: cli-wrap/src/main/resources/cli/pretty.py [hunk 5]
```diff
@@ -125,6 +125,69 @@ def printReportAlign(data):
...
+
+    if len(data['unchanged']) == 0:
+        print "unchanged : NONE"
+    else:
+        print "unchanged :"
+        for module in data['unchanged']:
+            print "    groupId : " + module['groupId'] 
+            print "    artifactId : " + module['artifactId']
+            print "    version : " + module['version']
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/pretty.py]: Here isn't blankline separator?

File: cli-wrap/src/main/resources/cli/pretty.py (hunk 4 above)
Comment (janinko) [inline cli-wrap/src/main/resources/cli/pretty.py]: And maybe there could be something like:
```
1.2.3.redhat-4     ->    1.2.4.redhat-2
````

**Pull request closed without merging**